From e1295099f8b42670718ba875cb6749a90042293f Mon Sep 17 00:00:00 2001
From: Zentaro Kavanagh <zentaro@chromium.org>
Date: Thu, 14 Jun 2018 13:13:37 -0700
Subject: [PATCH] Fix configure to support ncurses w/ tinfo

- The existing code checked for both ncursesw and ncurses and if
  both were not found, NCURSES_LIBS was not set correctly.
- Removed redundant concatenation to $LIBS since the makefile.am
  already maps NCURSES_LIBS into LIBS.
- Patch sent upstream to powertop mailing list [1]

[1] - https://lists.01.org/pipermail/powertop/2018-June/002021.html
---
 configure.ac | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index d6a15e1..c6ee50a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -110,10 +110,13 @@ AC_CHECK_FUNCS([ \
 
 AC_SEARCH_LIBS([clock_gettime], [rt])
 
-PKG_CHECK_MODULES([NCURSES], [ncursesw ncurses], [LIBS="$LIBS $ncurses_LIBS"], [
-	AC_SEARCH_LIBS([delwin], [ncursesw ncurses], [], [
-		AC_MSG_ERROR([ncurses is required but was not found])
-	], [])
+PKG_CHECK_MODULES([ncursesw], [ncursesw],
+	[NCURSES_CFLAGS="$ncursesw_CFLAGS"; NCURSES_LIBS="$ncursesw_LIBS"], [
+	PKG_CHECK_MODULES([NCURSES], [ncurses], [], [
+		AC_SEARCH_LIBS([delwin], [ncursesw ncurses], [], [
+			AC_MSG_ERROR([ncurses is required but was not found])
+		])
+	])
 ])
 
 has_libpci=0
-- 
2.18.0.rc1.242.g61856ae69a-goog

